src/main/java/tigase/xmpp/XMPPIOService.java

```diff
@@ -328,12 +331,24 @@ public long getTotalPacketsSent() {
 		return totalPacketsSent;
 	}
 
+	public Optional<JID> getAuthorisedUserJid() {
+		return Optional.ofNullable(authorisedUserJid);
+	}
+
+	public void setAuthorisedUserJid(JID authorisedUserJid) {
+		this.authorisedUserJid = authorisedUserJid;
+	}
+
+	@Deprecated
+	@TigaseDeprecated(removeIn = "9.0.0", since = "8.2.0", note = "#getAuthorisedUserJid should be used instead")
 	public String getUserJid() {
-		return this.jid;
+		return getAuthorisedUserJid().isPresent() ? getAuthorisedUserJid().get().toString() : null;
 	}
 
+	@Deprecated
+	@TigaseDeprecated(removeIn = "9.0.0", since = "8.2.0", note = "#setAuthorisedUserJid should be used instead")
 	public void setUserJid(String jid) {
-		this.jid = jid;
+		this.authorisedUserJid = JID.jidInstanceNS(jid);
```

Comment:
As far as I recall, this `jid` (string) is passed to the connection manager from the session manager and it is just a bare JID. Please verify if that is full jid or just bare jid.

Reply:
I've done it before creating PR. After auth but before bind it's just BareJID (obviously) but after binding it's proper FullJID. Otherwise this wouldn't have worked...

e.g.:
```
[2021-09-01 16:12:40:362] [FINEST  ] [  in_28-message-router ] MessageRouter.processPacket()    : 1. Packet will be processed by: sess-man@localhost, from=c2s@localhost/0:0:0:0:0:0:0:1_5222_0:0:0:0:0:0:0:1_61634, to=sess-man@localhost, serverAuthorisedStanzaFrom=Optional[admin@atlantiscity/1204607085-tigase-1], DATA=<iq id="aacaa" type="get" xmlns="jabber:client"><query xmlns="jabber:iq:private"><roster xmlns="roster:delimiter"/></query></iq>, SIZE=132, XMLNS=jabber:client, PRIORITY=NORMAL, PERMISSION=NONE, TYPE=get, STABLE_ID=null
```

Reply:
Then this is OK.